File: 1991.py
def preorder(x):
    if x == -1:
        return
    print(chr(x + ord('A')), end="")
    preorder(lst[x][0])
    preorder(lst[x][1])

def inorder(x):
    if x == -1:
        return
    inorder(lst[x][0])
    print(chr(x + ord('A')), end="")
    inorder(lst[x][1])

def postorder(x):
    if x == -1:
        return
    postorder(lst[x][0])
    postorder(lst[x][1])
    print(chr(x + ord('A')), end="")

t = int(input())
lst = [[-1]*2 for i in range(26)]

for i in range(t):
    ch1, ch2, ch3 = map(str, input().split())
    lst[ord(ch1) - ord('A')][0] = ord(ch2) - ord('A')
    lst[ord(ch1) - ord('A')][1] = ord(ch3) - ord('A')
    if ch2 == '.':
        lst[ord(ch1) - ord('A')][0] = -1
    if ch3 == '.':
        lst[ord(ch1) - ord('A')][1] = -1

preorder(0)
print()
inorder(0)
print()
postorder(0)
print()

File: 1991.cpp
#include <iostream>
using namespace std;
int node[50][2];

void preorder(int x) {
    if(x == -1) return;
    printf("%c", x + 'A');
    preorder(node[x][0]);
    preorder(node[x][1]);
}

void inorder(int x) {
    if(x == -1) return;
    inorder(node[x][0]);
    printf("%c", x + 'A');
    inorder(node[x][1]);
}

void postorder(int x) {
    if(x == -1) return;
    postorder(node[x][0]);
    postorder(node[x][1]);
    printf("%c", x + 'A');
}

int main() {
    int n;
    cin >> n;
    for(int i = 0; i < n; i++) {
        char x, y, z;
        cin >> x >> y >> z;
        x = x - 'A';
        if(y == '.') {
            node[x][0] = -1;
        } else {
            node[x][0] = y - 'A'; 
        }
        if(z == '.') {
            node[x][1] = -1;
        } else {
            node[x][1] = z - 'A';
        }
    }
    preorder(0);
    printf("\n");
    inorder(0);
    printf("\n");
    postorder(0);
    printf("\n");
    return 0;
}